src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/CreateAccount/CreateAccountCommandHandler.cs

```diff
@@ -117,42 +117,24 @@ public class CreateAccountCommandHandler : IAsyncRequestHandler<CreateAccountCom
             await _mediator.SendAsync(new PublishGenericEventCommand { Event = genericEvent });
         }
 
-        private async Task AddPayeSchemes(CreateAccountCommand message, string[] emprefs, CreateAccountResult returnValue)
+        private async Task AddPayeSchemes(CreateAccountCommand message, string empref, CreateAccountResult returnValue)
         {
-            if (emprefs.Length > 1)
-            {
-                for (var i = 1; i < emprefs.Length; i++)
-                {
-                    await _accountRepository.AddPayeToAccount(new Paye { AccountId = returnValue.AccountId, EmpRef = emprefs[i], AccessToken = message.AccessToken, RefreshToken = message.RefreshToken });
-                }
-            }
+            await _accountRepository.AddPayeToAccount(new Paye { AccountId = returnValue.AccountId, EmpRef = empref, AccessToken = message.AccessToken, RefreshToken = message.RefreshToken });
         }
 
-        private async Task RefreshLevy(CreateAccountResult returnValue, string[] emprefs)
+        private async Task RefreshLevy(CreateAccountResult returnValue, string empref)
         {
-            foreach (var empref in emprefs)
-            {
-                await _refreshEmployerLevyService.QueueRefreshLevyMessage(returnValue.AccountId, empref);
-            }
+            await _refreshEmployerLevyService.QueueRefreshLevyMessage(returnValue.AccountId, empref);
         }
 
-        private async Task PublishAddPayeSchemeMessage(IEnumerable<string> emprefs)
+        private async Task PublishAddPayeSchemeMessage(string empref, long accountId, string signedByName)

```

Comment:
Messages are still using signedByName parameters names. These need to be changed otherwise they will confuse people. Should be createdByName as not all messages have relations to signing agreement